frontend/src/app/worker/import.cljs

```diff
@@ -162,25 +162,26 @@
         pages (->> (:pages file-desc)
                    (mapv #(vector % (get-in index [(keyword %) :name]))))]
     (->> (rx/from pages)
-         (rx/flat-map #(process-page file-id zip %))
+         (rx/mapcat #(process-page file-id zip %))
          (merge-reduce import-page file)
          (rx/flat-map send-changes)
          (rx/ignore))))
 
+(defn process-package
+  [project-id zip-file]
+  (->> (uz/get-file zip-file "manifest.json")
+       (rx/flat-map (comp :files json/decode :content))
```

Comment:
If the manifest file does not exist, or has an incorrect value, does this have a controlled behavior?

Reply:
Yes, will return error at the end of the stream

Reply:
In any case I'll handle error management in a future PR

Reply:
Ok. We must be extremely cautious with all this, it may be a main security hole if not well protected.

For example, check the case of manifest.json exists and is a valid json file, but does not contain any :name or has a malicious file name in it...